agent/provider/src/cli/keystore.rs

```diff
@@ -67,9 +68,15 @@ fn add(config: ProviderConfig, add: Add) -> anyhow::Result<()> {
             }
         }
         KeystoreLoadResult::NothingNewToLoad { skipped } => {
-            println_conditional(&config, "No new certificate to add. Skipped:");
+            println_conditional(&config, "No new certificate to add.");
+            println_conditional(&config, "Dropped duplicated certificates:");
             let certs_data = util::to_cert_data(&skipped)?;
-            print_cert_list(&config, certs_data)?;
+            if !config.json {
+                print_cert_list(&config, certs_data)?;
+            } else {
+                // no new certificate added, so empty list for json output
+                print_cert_list(&config, Vec::new())?;
+            }
```

Comment:
Wouldn't it be clearer to just do
```rust
let certs_data = util::to_cert_data(&skipped)?;
if !config.json {
    println!("No new certificate to add.")
    println!("Dropped duplicated certificates:")
    print_cert_list(&config, certs_data)?;
} else {
    // no new certificate added, so empty list for json output
    print_cert_list(&config, Vec::new())?;
}
```

Reply:
To some degree. I wanted to use `println_conditional` for sake of consistency - without it these `println`s are suddenly special, but I will change it.
I am not exactly proud with the way it ended up - I have been tweaking printed output after implementing it to make it more clear for the user  (adding these `skipped` lists everywhere), so it is dotted with conditional statements and tuples you rightly complained on later.
Done.